### DISCUSSION TITLE
Support for Migrating Away from SmartThings

### DISCUSSION THREAD

**Discussion by jflowers** [zigpy/zigpy-znp#169] [Q&A] [answered]
Is it possible to perform a backup from a SmartThings hub to restore to zigbee2mqtt?

**Comment by puddly** [comment, marked as answer]
Not with this package.  You can, however, use a Zigbee packet sniffer with Wireshark and join any device to your SmartThings hub to capture all of the relevant network information.

With that, you can manually fill in a zigpy backup JSON, [restore it to a Z-Stack radio, and finally take a Z2M-compatible backup](https://github.com/zigpy/zigpy-cli/#network-backup) to replace Z2M's `coordinator_backup.json` with (make sure to also update Z2M's config with these same settings, it will erase your settings if the two don't match up):

```json
{
    "backup_time": "2022-08-31T16:21:23.453068+00:00",
    "node_info": {
        "nwk": "0000",
        "ieee": "00:3c:84:ff:fe:92:bb:ef",  // fill this in
        "logical_type": "coordinator"
    },
    "network_info": {
        "extended_pan_id": "6b:6f:94:73:1d:6b:63:a8",  // and this
        "pan_id": "6542",  // and this
        "nwk_update_id": 0,  // and this
        "nwk_manager_id": "0000",
        "channel": 15,  // and this
        "channel_mask": [15],
        "security_level": 5,
        "network_key": {
            "key": "51:6d:19:3b:e5:7a:af:8c:4c:ec:64:82:e2:a1:5e:f6",  // and this
            "tx_counter": 81921,  // and this
            "rx_counter": 0,
            "seq": 0,
            "partner_ieee": "ff:ff:ff:ff:ff:ff:ff:ff"  // copy `node_info.ieee`
        },
        "tc_link_key": {
            "key": "5a:69:67:42:65:65:41:6c:6c:69:61:6e:63:65:30:39",
            "tx_counter": 0,
            "rx_counter": 0,
            "seq": 0,
            "partner_ieee": "00:3c:84:ff:fe:92:bb:ef"  // copy `node_info.ieee`
        },
        "key_table": [],
        "children": [],
        "nwk_addresses": {},
        "stack_specific": {},
        "metadata": {},
        "source": "manual"
    }
}
```

If you want to try this, feel free to post the packet capture here and I can try to help out.

**Reply by jflowers** [reply]
thanks, need to go research zigbee packet sniffer now

**Reply by puddly** [reply]
Most Zigbee radios support packet capture, though most also require alternate firmware (with the exception of Silicon Labs EmberZNet radios).  You can use any Conbee, Silicon Labs radio, or even a CC2652 if you have a TI dev kit or another way to re-flash the bootloader. Or a dedicated nRF52840 stick.

**Reply by tube0013** [reply]
I like the nRF52840 as a standalone sniffer, relatively cheap and easy to set up with wireshark

<https://infocenter.nordicsemi.com/index.jsp?topic=%2Fug_sniffer_802154%2FUG%2Fsniffer_802154%2Fintro_802154.html>

**Reply by jflowers** [reply]
thanks again, it'll be delivered on Friday
